src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.hxx

```diff
@@ -0,0 +1,280 @@
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_geometry_occ.hxx
+ * This gometry implements OpenCASCADE geometries. It enables the option to link different 
+ * 1 and 2 dimensional occ geometries to the edges and faces of refinement trees. 
+ * The geometry of the refinement tree gets bent accordingly.
+ */
+
+#ifndef T8_GEOMETRY_OCC_HXX
+#define T8_GEOMETRY_OCC_HXX
+
+#define T8_WITH_OCC 1
+#if T8_WITH_OCC
+
+#include <t8.h>
+#include <t8_geometry/t8_geometry_base.hxx>
+#include <t8_cmesh/t8_cmesh_types.h>
+
+#include <TopoDS_Shape.hxx>
+#include <TopExp.hxx>
+#include <gp_Pnt.hxx>
+#include <Geom_Curve.hxx>
+#include <Geom_Surface.hxx>
+
+/* The vertices of each edge of a hexahedron. Used in the occ geometry. */
+extern const int    t8_edge_vertex_to_tree_vertex[T8_ECLASS_MAX_EDGES][2];
+
+/* The faces connected to each edge. */
+extern const int    t8_edge_to_face[T8_ECLASS_MAX_EDGES][2];
+
+/* The edges of a face to the edges of a tree */
+extern const int
+     t8_face_edge_to_tree_edge[T8_ECLASS_MAX_FACES][T8_ECLASS_MAX_EDGES_2D];
+
+/**
+ * Definition of an occ geometry function.
+ * This function maps reference coordinates to physical
+ * coordinates regarding the occ geometries linked to the cells edges and faces.
+ * \param [in]  cmesh       The cmesh.
+ * \param [in]  gtreeid     The global tree (of the cmesh) in which the reference point is.
+ * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+ * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords.
+ * \param [in]  tree_data   The data of the current tree as loaded by a \ref t8_geom_load_tree_data_fn.
+ * \param [in]  user_data   The user data pointer stored in the geometry.
+ */
+typedef void        (*t8_geom_occ_fn) (t8_cmesh_t cmesh,
+                                       t8_gloidx_t gtreeid,
+                                       const double *ref_coords,
+                                       double out_coords[3],
+                                       const void *tree_data,
+                                       const void *user_data);
```

Comment:
Is this typedef required?
i do not see where it is used.

Reply:
resolved in 0456f9c2